controllers/front/post.php

```diff
@@ -94,6 +94,14 @@ public function initContent()
 
         $this->context->smarty->assign($postProperties);
 
+        $hookHeader = $this->context->smarty->getTemplateVars('HOOK_HEADER');
+        $hookHeader .= '<meta property="og:url" content="'.\Tools::getHttpHost(true).$_SERVER['REQUEST_URI'].'" >'.PHP_EOL;
+        $hookHeader .= '<meta property="og:type" content="article" >'.PHP_EOL;
+        $hookHeader .= '<meta property="og:title" content="'.$post->meta_title.' - '.Configuration::get('PS_SHOP_NAME').'" >'.PHP_EOL;
+        $hookHeader .= '<meta property="og:image" content="'.\Tools::getHttpHost(true).'/img/beesblog/posts/'.(int) $post->id.'.jpg'.'" >'.PHP_EOL;
+        $hookHeader .= '<meta property="og:description" content="'.$post->meta_description.'" >'.PHP_EOL;        
+        $this->context->smarty->assign('HOOK_HEADER' , $hookHeader);        
+        
```

Comment:
Could you please extract this block to `displayHeader` hook? That hooks is a standard way to include additional header metadata. And you don't have to redefine existing template var 'HOOK_HEADER'. . Please make sure that hook returns information for blog pages only.

Another good reason for having that functionality extracted in hook is to allow merchants to unhook it, in case their theme already emits open graph metadata. A lot of themes do contain some metadata already, and this code has a potential to break them.

For reference: 
- https://github.com/thirtybees/socialsharing/blob/08545935e5286fd3d4f46eefc51c46734a22502f/socialsharing.php#L79 
- https://github.com/thirtybees/socialsharing/blob/08545935e5286fd3d4f46eefc51c46734a22502f/socialsharing.php#L219
- https://github.com/thirtybees/socialsharing/blob/master/views/templates/hook/socialsharing_header.tpl